nexus/src/app/certificate.rs

```diff
@@ -0,0 +1,230 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! x.509 Certificates
+
+use crate::context::OpContext;
+use crate::db;
+use crate::db::lookup;
+use crate::db::lookup::LookupPath;
+use crate::db::model::Name;
+use crate::external_api::params;
+use omicron_common::api::external::CreateResult;
+use omicron_common::api::external::DataPageParams;
+use omicron_common::api::external::DeleteResult;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::ListResultVec;
+use omicron_common::api::external::NameOrId;
+use openssl::pkey::PKey;
+use openssl::x509::X509;
+use ref_cast::RefCast;
+use std::sync::Arc;
+use uuid::Uuid;
+
+fn validate_certs(input: Vec<u8>) -> Result<(), String> {
+    let certs = X509::stack_from_pem(&input.as_slice())
+        .map_err(|err| format!("Failed to parse certificate as PEM: {err}"))?;
+    if certs.is_empty() {
+        return Err("could not parse".to_string());
```

Comment:
This failure mode seems rather undebuggable...I guess we don't get any more information than this out of the library?

Reply:
Nice!